synthesizer/process/src/verify_execution.rs

```diff
@@ -104,13 +105,23 @@ impl<N: Network> Process<N> {
             // Ensure each output is valid.
             let num_inputs = transition.inputs().len();
             let num_outputs = transition.outputs().len();
-            if transition
-                .outputs()
-                .iter()
-                .enumerate()
-                .any(|(index, output)| !output.verify(function_id, transition.tcm(), num_inputs + index))
-            {
-                bail!("Failed to verify a transition output")
+            for (index, output) in transition.outputs().iter().enumerate() {
+                // If the consensus version are before `ConsensusVersion::V8`, ensure the output record is on Version 0.
+                // if the consensus version is on or after `ConsensusVersion::V8`, ensure the output record is on Version 1.
+                if let Some((_, record)) = output.record() {
+                    if (ConsensusVersion::V1..=ConsensusVersion::V7).contains(&consensus_version) {
```

Comment:
Just double checking if there's not an off-by-one here. On V8, we still expect the old inclusion version, but new records? https://github.com/ProvableHQ/snarkVM/pull/2792/files#diff-c84253a64c2536d7171b14e7867f0224cd9beef2193a1fa6c87868fce5d07dfbR160

Reply:
I see now the logic is the same everywhere already. **after** block height X, `ConsensusVersion::V8` and all of its rules come into effect, so the first block height from which it is enforced is effectively `X+1`.